### PR TITLE
Use ObjectFactory.directoryProperty() if possible

### LINKED ISSUE #64: GitPublishReset uses deprecated ProjectLayout.directoryProperty() method
With Gradle 5.0, getting the following warning that is coming from this plugin:
```
The ProjectLayout.directoryProperty() method has been deprecated. This is scheduled to be removed in Gradle 6.0. Please use the ObjectFactory.directoryProperty() method instead.
```

### PR DESCRIPTION
**Pull request by marcphilipp**
ProjectLayout.directoryProperty() was deprecated in Gradle 5.0 and
replaced with ObjectFactory.directoryProperty(). In order to stay
backward compatible with Gradle 4.x, the latter is now called first and
the former is used as a fallback.

Resolves #64.

### CODE REVIEW

**Review by sormuras — APPROVED** [review]
LGTM

File: src/main/java/org/ajoberstar/gradle/git/publish/tasks/DirectoryPropertyFactory.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public static DirectoryProperty create(Project project) {
+        return create(project.getLayout(), project.getObjects());
+    }
+
+    public static DirectoryProperty create(ProjectLayout layout, ObjectFactory objectFactory) {
+        try {
+            return objectFactory.directoryProperty();
+        } catch (NoSuchMethodError e) {
```
Comment (ajoberstar) [inline src/main/java/org/ajoberstar/gradle/git/publish/tasks/DirectoryPropertyFactory.java:23]: We could avoid the exception by using `if (GradleVersion.current() < GradleVersion.version('5.0'))` or something similar, right?

Comment (marcphilipp) [reply src/main/java/org/ajoberstar/gradle/git/publish/tasks/DirectoryPropertyFactory.java:23]: Right. Would you prefer that?

Comment (ajoberstar) [reply src/main/java/org/ajoberstar/gradle/git/publish/tasks/DirectoryPropertyFactory.java:23]: Yeah. If you don't have time, let me know. I plan to make some updates to the plug-in later today.

**Comment by ajoberstar** [comment]
I have a fix released in 2.1.0-beta.1. Thanks for sending this in, nudged me to get some other fixes done.

**Pull request closed without merging**